Extension/src/LanguageServer/cppBuildTaskProvider.ts

```diff
@@ -0,0 +1,357 @@
+			 new CustomBuildTaskTerminal(resolvedcompilerPath, args, options, target.name)
+            ), isCl ? '$msCompile' : '$gcc');
+
+        task.group = vscode.TaskGroup.Build;
+
+        return task;
+    };
+
+    async ensureBuildTaskExists(taskName: string): Promise<void> {
+        const rawTasksJson: any = await getRawTasksJson();
+
+        // Ensure that the task exists in the user's task.json. Task will not be found otherwise.
+        if (!rawTasksJson.tasks) {
+            rawTasksJson.tasks = new Array();
+        }
+        // Find or create the task which should be created based on the selected "debug configuration".
+        let selectedTask: vscode.Task | undefined = rawTasksJson.tasks.find((task: any) => task.label && task.label === taskName);
+        if (selectedTask) {
+            return;
+        }
+
+        const buildTasks: vscode.Task[] = await this.getTasks(true);
+        selectedTask = buildTasks.find(task => task.name === taskName);
+        console.assert(selectedTask);
+        if (!selectedTask) {
+            throw new Error("Failed to get selectedTask in ensureBuildTaskExists()");
+        }
+
+        rawTasksJson.version = "2.0.0";
+
+        const selectedTask2: vscode.Task = selectedTask;
+        if (!rawTasksJson.tasks.find((task: any) => task.label === selectedTask2.definition.label)) {
+            const task: any = {
+                ...selectedTask2.definition,
+                problemMatcher: selectedTask2.problemMatchers,
+                group: { kind: "build", "isDefault": true }
+            };
+            rawTasksJson.tasks.push(task);
+        }
+
+        // TODO: It's dangerous to overwrite this file. We could be wiping out comments.
+        const settings: OtherSettings = new OtherSettings();
+        const tasksJsonPath: string | undefined = getTasksJsonPath();
+        if (!tasksJsonPath) {
+            throw new Error("Failed to get tasksJsonPath in ensureBuildTaskExists()");
+        }
+
+        await util.writeFileText(tasksJsonPath, JSON.stringify(rawTasksJson, null, settings.editorTabSize));
+    }
+}
+
+class CustomBuildTaskTerminal implements vscode.Pseudoterminal {
+    private writeEmitter  = new vscode.EventEmitter<string>();
+    private closeEmitter = new vscode.EventEmitter<number>();
+    public get onDidWrite(): vscode.Event<string> { return this.writeEmitter.event; }
+    public get onDidClose(): vscode.Event<number> { return this.closeEmitter.event; }
+
+    private fileWatcher: vscode.FileSystemWatcher | undefined;
+
+
+    constructor(private command: string, private args: string[], private options: cp.ExecOptions | undefined, private workspaceRoot: string) {
+    }
+
+
+    open(_initialDimensions: vscode.TerminalDimensions | undefined): void {
+        telemetry.logLanguageServerEvent("cppBuildTaskStarted");
+        const pattern: string = path.join(this.workspaceRoot, 'cppBuild');
```

Comment:
This file watcher stuff appears to be an artifact of the extension sample.  I think it can be safely removed.